plutus/budget_manager/budget_manager.py

```diff
@@ -0,0 +1,227 @@
+
+        else:
+            log.error("Project config verification failed.")
+            metrics.incr("error_count", tags = ["type:misconfig", f"project_id:{project_id}", f"config_type:{config_type}"])
+            sys.exit(1)
+
+
+    # Iterate over all configured parent folder ids
+    for parent_dict in budget_dict['parent_folders']:
+        parent_id = parent_dict['parent_folder_id']
+        config_type = PLUTUS_CONFIG_TYPE_PARENT
+
+        if verify_parent_yaml(parent_dict):
+            parent_filter = {'parent.id': parent_id}
+
+            for p in resource_manager_client.list_projects(parent_filter):
+                # Every iteration we overwrite the 'project_id' key for each project under parent folder
+                parent_dict['project_id'] = p.project_id
+                project = ProjectBudget(parent_dict, config_type, billing_account_id, default_pubsub_topic)
+
+                budget = gcp.get_and_update_or_create_budget(project)
+
+                if budget is not None:
+                    with mysql_conn.cursor() as mysql_cursor:
+                        upsert_budget(mysql_cursor, budget, project.project_id, config_type, project.alert_emails)
+
+        else:
+            log.error("Parent folder config verification failed.")
+            metrics.incr("error_count", tags = ["type:misconfig", f"parent_id:{parent_id}", f"config_type:{config_type}"])
+            sys.exit(1)
+
+
+    # TODO - Have a google support ticket, currently specifying more than one label returns a union not an intersection
+    for label_dict in budget_dict['labels']:
+        config_type = PLUTUS_CONFIG_TYPE_LABEL
+
+        if verify_labels_yaml(label_dict):
+
+            labels_filter = {}
+            for row in label_dict['label_list']:
+                for key in row:
+                    labels_filter[f"labels.{key}"] = row[key]
+
+            # TODO - remove this count once google support ticket resolved
+            count = 0
+            for p in resource_manager_client.list_projects(filter_params=labels_filter):
+                # TODO - rm me
+                count += 1
+
+                # Every iteration, we overwrite the 'project_id' key for each project that matches labels
+                label_dict['project_id'] = p.project_id
+                project = ProjectBudget(label_dict, config_type, billing_account_id, default_pubsub_topic)
+
+                # TODO - add back more labels to config and test once google ticket resolved
+                budget = gcp.get_and_update_or_create_budget(project)
+
+                if budget is not None:
+                    with mysql_conn.cursor() as mysql_cursor:
+                        upsert_budget(mysql_cursor, budget, project.project_id, config_type, project.alert_emails)
+
+            # TODO - remove this after ticket resolved
+            log.info(count)
+
+        else:
+            log.error("Labels config verification failed.")
```

Comment:
There are a couple areas in the code where we try to handle a mis-configuration. I think we should define an custom exception class and raise that exception when we run into that issue. As part of the exception we could also include logging and metrics increment.